custom/icds_reports/static/js/directives/mobile-datepicker/mobile-datepicker.directive.js

```diff
@@ -309,22 +310,60 @@ angular.module('icdsApp').directive('mobileDatepicker', function() {
 
             })(jQuery);
 
+            //this method returns months to be displayed corresponding to the year selected
+            // if its minYear
+            getItems = function () {
+                return ($scope.date.getFullYear() === $scope.minYear) ? months.slice($scope.startMonth - 1) :
+                        (($scope.date.getFullYear() === new Date().getFullYear()) ?
+                            months.slice(0, new Date().getMonth()+1) : months);
+            };
+
+            // this method returns which month to be shown at the center of datepicker
+            // it is picked from items array. But
+            getCenter = function () {
+                let centerMonth = $scope.date.getMonth();
+                if ($scope.date.getFullYear() === $scope.minYear) {
+                    centerMonth = $scope.date.getMonth()- ($scope.startMonth - 1);
+                } else if ($scope.date.getFullYear() === new Date().getFullYear()) {
+                    if ($scope.date.getMonth() > new Date().getMonth()) {
+                        centerMonth = new Date().getMonth();
+                    }
+                }
+                return centerMonth;
+            };
 
             var today = $scope.date;
             var months = ['January','February','March','April','May','June','July','August','September','October','November','December'];
-            let center = $scope.date.getMonth();
-            if ($scope.date.getFullYear() === 2017) {
-                center = $scope.date.getMonth()-2;
-            } else if ($scope.date.getFullYear() === new Date().getFullYear()) {
-                if ($scope.date.getMonth() > new Date().getMonth()) {
-                    center = new Date().getMonth();
+
+            //this method returns months to be displayed corresponding to the year selected
+            // if its minYear, then from months list months before startMonth is removed
+            // if its current year then from months list months that comes after the current month are removed
+            getItems = function () {
```

Comment:
is there a reason you define both of these functions twice? Are they different? (I suspect one set should be deleted)